statics/js/components/preference.js

```diff
@@ -0,0 +1,133 @@
+              <input class="form-control" v-model="filter.filter"/>\
+            </div>\
+          </div>\
+        </div>\
+        <button class="btn btn-primary btn-round-xs btn-xs" type="button" @click="addFilter">+</button>\
+      </div>\
+      <div class="form-group">\
+        <label for="bw-threshold">Bandwidth Threshold</label>\
+        <a><i class="fa fa-question help-text" aria-hidden="true" title="bandwidth threshold mode (relative/absolute)"></i></a>\
+        <input id="bw-threshold" type="text" class="form-control input-sm" v-model="bwThreshold"/>\
+      </div>\
+      <div class="form-group">\
+        <label for="bw-update-interval">Bandwidth Update Interval</label>\
+        <a><i class="fa fa-question help-text" aria-hidden="true" title="bandwidth udadate interval(in seconds)"></i></a>\
+        <input id="bw-update-interval" type="number" class="form-control input-sm" v-model="bwUpdatePeriod"/>\
+      </div>\
+      <div class="form-group">\
+        <label for="bw-abs-active">Bandwidth Absolute Active</label>\
+        <input id="bw-abs-active" type="number" class="form-control input-sm" v-model="bwAbsActive" min="0"/>\
+      </div>\
+      <div class="form-group">\
+        <label for="bw-abs-warning">Bandwidth Absolute Warning</label>\
+        <input id="bw-abs-warning" type="number" class="form-control input-sm" v-model="bwAbsWarning" min="0"/>\
+      </div>\
+      <div class="form-group">\
+        <label for="bw-abs-alert">Bandwidth Absolute Alert</label>\
+        <input id="bw-abs-alert" type="number" class="form-control input-sm" v-model="bwAbsAlert" min="0"/>\
+      </div>\
+      <div class="form-group">\
+        <label for="bw-rel-active">Bandwidth Relative Active</label>\
+        <input id="bw-rel-active" type="number" class="form-control input-sm" v-model="bwRelActive" min="0" max="1" step="0.1"/>\
+      </div>\
+      <div class="form-group">\
+        <label for="bw-rel-warning">Bandwidth Relative Warning</label>\
+        <input id="bw-rel-warning" type="number" class="form-control input-sm" v-model="bwRelWarning" min="0" max="1" step="0.1"/>\
+      </div>\
+      <div class="form-group">\
+        <label for="bw-rel-alert">Bandwidth Relative Alert</label>\
+        <input id="bw-rel-alert" type="number" class="form-control input-sm" v-model="bwRelAlert" min="0" max="1" step="0.1"/>\
+      </div>\
+      <div class="button-holder row">\
+        <button class="btn btn-lg btn-primary" type="submit">Save</button>\
+        <button class="btn btn-lg btn-danger" type="button" @click="cancel"> Cancel</button>\
+      </div>\
+    </form>\
+  ',
+
+  data: function() {
+    var filters = [{name:"", filter:""}];
+    
+    f = JSON.parse(localStorage.getItem("filters"));
+    if (f && f.length > 0) filters = f;
+
+    return {
+      collapseIndex: localStorage.collapseIndex,
+      filters: filters,
+      bwThreshold: localStorage.bandwidthThreshold,
```

Comment:
We should return the default values from config file is provided

Reply:
if we pass a config value it will be changing the local value every time we go to preference pane.